sphinx_gallery/gen_rst.py

```diff
@@ -440,12 +465,28 @@ def generate_dir_rst(src_dir, target_dir, gallery_conf, seen_backrefs):
         ))
         findex.write(subsection_index_content)
 
-        # add toctree to file only if toctree is not empty
-        if len(subsection_toctree_filenames) > 0:
+        # Create toctree for index file
+        # with all gallery items which belong to current subsection
+        # and add it to generated index rst file if need be.
+        # Toctree cannot be empty
+        # and won't be added if include_toctree is false
+        # (this is useful when generating the example gallery's main
+        # index rst file, which should contain only one toctree)
+        if len(subsection_toctree_filenames) > 0 and include_toctree:
+            subsection_index_toctree = """
+.. toctree::
+   :hidden:
+
+   %s\n
+""" % "\n   ".join(subsection_toctree_filenames)
             findex.write(subsection_index_toctree)
 
-    return subsection_index_content, costs, subsection_index_toctree, \
-        subsection_index_path
+    return (
+        subsection_index_path,
+        subsection_index_content,
+        costs,
+        subsection_toctree_filenames,
+    )
```

Comment:
Can you restore the old order? Technically the API is public

https://sphinx-gallery.github.io/stable/gen_modules/sphinx_gallery.gen_rst.html#sphinx_gallery.gen_rst.generate_dir_rst

So we should maybe change it as little as possible. If you restore the old order and args here and add a `return_toctree_filenames=False` default kwarg (which, when True, means you put `subsection_toctree_filenames` in place of or in addition to the `subsection_index_toctree` that is returned when False) then I think we stay backward compatible.

Reply:
Oh... in that case, maybe don't bother :) Up to you. If nobody complained so far and we already broke the rule about API stability, breaking it again seems like less of a problem... I'm not sure we really explicitly guarantee a stable API anyway. I just usually try to avoid changing it if it doesn't add too much additional complexity.

Reply:
Maybe it's silly of me, but since the docstring was so concise, I just assumed no one would be using it (I thought it's hard to use a method which is not documented haha)